app/lambda_function.py

```diff
@@ -1,25 +1,111 @@
 import json
+import globus_sdk
+import boto3
 
+from time import sleep
 from doi import call_datacite
 from tiny_router import TinyLambdaRouter
 
+# garden-dev index
+GARDEN_INDEX_UUID = "58e4df29-4492-4e7d-9317-b27eba62a911"
+
 app = TinyLambdaRouter()
 
 
+def get_secret():
+    secret_name = (
+        "arn:aws:secretsmanager:us-east-1:557062710055:secret:garden/globus_api-2YYuTW"
+    )
+    region_name = "us-east-1"
+
+    # Create a Secrets Manager client
+    session = boto3.session.Session()
+
+    client = session.client(service_name="secretsmanager", region_name=region_name)
+
+    get_secret_value_response = client.get_secret_value(SecretId=secret_name)
+    return eval(get_secret_value_response["SecretString"])
+
+
 def lambda_handler(event, context):
     return app.run(event, context)
 
 
-@app.route('/hello-world', methods=['GET'])
+@app.route("/hello-world", methods=["GET"])
 def hello(event, context, kwargs):
     return {
-        'statusCode': 200,
-        'body': json.dumps(
+        "statusCode": 200,
+        "body": json.dumps(
             {
                 "success": True,
-                'message': 'Hello there. You must be an authenticated Globus user.',
-            })
+                "message": "Hello there. You must be an authenticated Globus user.",
+            }
+        ),
     }
 
 
+@app.route("/garden-search-record", methods=["POST"])
+def publish(event, context, kwargs):
```

Comment:
Can we pull the `publish` implementation and `get_secret` into a separate file? I'd like to keep this main file mostly just the routing setup.